### ISSUE TITLE
When using private key authentication the key is required to have a passphrase

### ISSUE THREAD

**Issue by zchesty** [Snowflake-Labs/schemachange#52]
Sometimes a private key may not have a passphrase for it. schemachange will not allow a private key without a passphrase to be used.

**Comment by sfc-gh-jhansen** [comment]
Hi there @zchesty, thanks for raising this issue. If you have a suggest fix to the issue, please submit a PR and I'd be happy to review it.

**Comment by jcollum** [comment]
If you can avoid a private key without a passphrase you definitely should.

**Comment by ladislav-sindelka** [comment]
```
    with open(os.environ["SNOWFLAKE_PRIVATE_KEY_PATH"], "rb") as key:
      p_key= serialization.load_pem_private_key(
          key.read(),
          password = os.environ['SNOWFLAKE_PRIVATE_KEY_PASSPHRASE'].encode(),
          backend = default_backend()
      )
```
changed to
if len(os.getenv('SNOWFLAKE_PRIVATE_KEY_PASSPHRASE', '')) > 0:
    private_key_password = os.environ['SNOWFLAKE_PRIVATE_KEY_PASSPHRASE'].encode()
else:
    private_key_password = None
with open(os.environ["SNOWFLAKE_PRIVATE_KEY_PATH"], "rb") as key:
  p_key= serialization.load_pem_private_key(
      key.read(),
      password = os.environ['SNOWFLAKE_PRIVATE_KEY_PASSPHRASE'].encode(),
      backend = default_backend()
  )
```

by setting password to None, the serialization uses no password and does not crash.

**Comment by ladislav-sindelka** [comment]
opened PR https://github.com/Snowflake-Labs/schemachange/pull/65 for this

**Comment by SlightThought** [comment]
Hey all, I believe this PR broke detection of the SNOWFLAKE_PRIVATE_KEY_PATH environment variable. When I define that env var, it errors out with "raise ValueError("Unable to find connection credentials for private key or password authentication")" on line 242. However, if I define SNOWFLAKE_PASSWORD, the cli is able to detect it and use it. I tried defining both at the same time, and it defaulted to using SNOWFLAKE_PASSWORD (expected behaviour, but showing that it is not an issue with discovery of my env vars).

**Comment by SlightThought** [comment]
Update regarding prior comment:

The PR did not break detection of the env var. It works in 2.9.4 (I pulled the repo and tested locally) but the latest version available on PyPi is 2.9.2. When using that version, I receive the error above. So I think the solution here is making sure 2.9.4 is available to pip install and reflects what is present in the repo as of now.

**Comment by sfc-gh-jhansen** [comment]
Publishing version 3.0.0 to PyPI shortly with this fix.
